FloripaPublicTransportationTests/RestApiTests.swift

```diff
@@ -0,0 +1,343 @@
+    
+    /// Generic function to be used by tests of `findDeparturesByRouteId` that use a valid JSON
+    /// - Parameter stubResponse: the block that returns an `OHHTTPStubsResponse` with stubbed NSData/NSDictionary
+    /// - Parameter completionTests: the block of tests to be executed against the returned array
+    private func testValidFindDeparturesByRouteId(stubResponse: OHHTTPStubsResponseBlock, completionTests: (departures: [Departure]) -> Void)
+    {
+        testFindDeparturesByRouteId(stubResponse) { departures in
+            XCTAssertNotNil(departures, "The returned array must not be nil")
+            completionTests(departures: departures!)
+        }
+    }
+    
+    /// Generic function to be used by tests of `findDeparturesByRouteId` that use an invalid JSON
+    /// (or none at all)
+    /// - Parameter stubResponse: the block that returns an `OHHTTPStubsResponse` with stubbed NSData/NSDictionary
+    private func testInvalidFindDeparturesByRouteId(stubResponse: OHHTTPStubsResponseBlock)
+    {
+        let nsLogArrayCountBefore = SystemLogAccessor.NSLogArray().count
+        testFindDeparturesByRouteId(stubResponse) { departures in
+            XCTAssertNotNil(departures, "The returned array must not be nil")
+            XCTAssertEqual(departures!.count, 0, "The returned array must be empty")
+            XCTAssertGreaterThan(SystemLogAccessor.NSLogArray().count, nsLogArrayCountBefore, "This application should have increased its number of logs due to the detected errors")
+            // @todo Consider verifying the messages instead of just the count
+        }
+    }
+    
+    /// Generic function to be used by any test of `findDeparturesByRouteId`
+    /// - Parameter stubResponse: the block that returns an `OHHTTPStubsResponse` with stubbed NSData/NSDictionary
+    /// - Parameter restApiCompletion: the `findDeparturesByRouteId` completion block
+    private func testFindDeparturesByRouteId(stubResponse: OHHTTPStubsResponseBlock, restApiCompletion: (departures: [Departure]?) -> Void)
+    {
+        stub(isHost("api.appglu.com") && isPath("/v1/queries/findDeparturesByRouteId/run"), response: stubResponse)
+        
+        var didExecuteCompletion = false
+        RestApi.findDeparturesByRouteId(7664576, completion: { departures in
+            restApiCompletion(departures: departures)
+            didExecuteCompletion = true
+        })
+        
+        // loop until the expectation is fulfilled:
+        waitForExpectationsWithTimeout(expectationTimeout) { error in
+            XCTAssertNil(error, "Expectation timeout")
+        }
+        
+        XCTAssertTrue(didExecuteCompletion, "The completion block must be executed")
+    }
+    
+    // MARK: - XCTestExpectation definitions
+    
+    private let expectationTimeout: NSTimeInterval = 5 // just a big enough timeout for the expectations
+    
+    private var expectation: XCTestExpectation?
+    
+    func onDone(){
+        expectation?.fulfill()
```

Comment:
Yes, as previously commented, testing an asynchronous methods require expectations to be waited until fulfilled. The onDone function allows me to fulfill it once the async method is completed.